packages/react/src/components/Listbox/ListboxContext.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { createContext, useContext, useMemo } from 'react';
+
+type UnknownElement<T> = T extends Element ? T : HTMLElement;
+type UnknownValue<T> = T extends string ? T : number;
+type ListboxOption<Element = HTMLElement, Value = string | number> = {
+  element: UnknownElement<Element>;
+  value?: UnknownValue<Value>;
+};
+
+type ListboxContext<T extends ListboxOption> = {
+  options: T[];
+  active: T | null;
+  selected: T | null;
+  setOptions: React.Dispatch<React.SetStateAction<T[]>>;
+  onSelect: (option: T) => void;
+};
+
+type ListboxProvider<T extends ListboxOption> = {
+  children: React.ReactNode;
+} & ListboxContext<T>;
+
+/* istanbul ignore next */
```

Comment:
> Why is typescript not happy here?
Sorry, I meant here.

Reply:
I'm still not clear what you're referring to. I do not have any typescript errors:

![vscode showing no errors on the referenced line](https://github.com/dequelabs/cauldron/assets/1062039/9b9fc697-39f5-42a7-ae49-2a078a10b942)

Reply:
nyc was complaining about none of the options getting invoked but I don't think we really care about these lines for coverage since it's stubbing out an empty context so I ignored these lines. ¯\\\_(ツ)_/¯